Makefile_packaging.mk

```diff
@@ -0,0 +1,244 @@
+# Common Makefile for including
+# Needs the following variables set at a minium:
+# NAME :=
+# DEB_NAME :=
+# SRC_EXT :=
+# SOURCE =
+# ID_LIKE := $(shell . /etc/os-release; echo $$ID_LIKE)
+
+ifeq ($(DEB_NAME),)
+DEB_NAME := $(NAME)
+endif
+ID_LIKE := $(shell . /etc/os-release; echo $$ID_LIKE)
+COMMON_RPM_ARGS := --define "%_topdir $$PWD/_topdir"
+DIST    := $(shell rpm $(COMMON_RPM_ARGS) --eval %{?dist})
+ifeq ($(DIST),)
+SED_EXPR := 1p
+else
+SED_EXPR := 1s/$(DIST)//p
+endif
+ifeq ($(SPEC),)
+SPEC    := $(NAME).spec
+endif
+ifeq ($(VERSION),)
+VERSION := $(shell rpm $(COMMON_RPM_ARGS) --specfile --qf '%{version}\n' $(SPEC) | sed -n '1p')
+endif
+DOT     := .
+DEB_VERS := $(subst rc,~rc,$(VERSION))
+DEB_RVERS := $(subst $(DOT),\$(DOT),$(DEB_VERS))
+DEB_BVERS := $(basename $(subst ~rc,$(DOT)rc,$(DEB_VERS)))
+RELEASE := $(shell rpm $(COMMON_RPM_ARGS) --specfile --qf '%{release}\n' $(SPEC) | sed -n '$(SED_EXPR)')
+ifeq ($(SRPM),)
+SRPM    := _topdir/SRPMS/$(NAME)-$(VERSION)-$(RELEASE)$(DIST).src.rpm
+endif
+ifeq ($(RPMS),)
+RPMS    := $(addsuffix .rpm,$(addprefix _topdir/RPMS/x86_64/,$(shell rpm --specfile $(SPEC))))
+endif
+DEB_TOP := _topdir/BUILD
+DEB_BUILD := $(DEB_TOP)/$(NAME)-$(DEB_VERS)
+DEB_TARBASE := $(DEB_TOP)/$(DEB_NAME)_$(DEB_VERS)
+ifeq ($(SOURCES),)
```

Comment:
`?=` does this.

Reply:
That does not behave the same way as := if the variable was not originally defined.  It would cause these variables to be expanded every time

Reply:
I'm not sure I'm following.  Isn't what you are trying to achieve this:

```
$ cat /tmp/Makefile
A := bar
A ?= foo

debug:
	echo $(A)
$ make -f /tmp/Makefile debug
echo bar
bar
$ sed -i -e 1d /tmp/Makefile
$ cat /tmp/Makefile
A ?= foo

debug:
	echo $(A)
$ make -f /tmp/Makefile debug
echo foo
foo
```
?

Reply:
$cat Makefile
A ?= $(shell sleep 1;date)

debug:
        @echo $(A)
        @echo $(A)

$make -f Makefile
Mon Jul 29 11:25:54 PDT 2019
Mon Jul 29 11:25:55 PDT 2019

The variables assigned with a := can not be changed to be assigned with a ?= or it causes them to expanded with each use, so that would be a change in behavior.

Reply:
Yes, `?=` behaves more like `=` than `:=`.  But often the distinction between using `=` and `:=` is not meaningful.  At times it is, but probably more times it is not.  Most times, `:=` is just used for performance of expanding only once instead of at each use.

Is the distinction meaningful in these cases?

Reply:
It was meaningful in cases where the information was potentially looked up from the spec file.
Not as meaningful as I have added the files from the src.rpm to this repo.